pyke/lightcurve.py

```diff
@@ -182,20 +189,38 @@ def get_lightcurve(self, flux_type, centroid_type='MOM_CENTR'):
                                     campaign=self.campaign,
                                     quarter=self.quarter,
                                     mission=self.mission,
-                                    cadenceno=self.cadenceno)
+                                    cadenceno=self.cadenceno,
+                                    id = self.hdu[0].header['KEPLERID'])
         else:
             raise KeyError("{} is not a valid flux type. Available types are: {}".
                            format(flux_type, self._flux_types))
 
-    def _quality_mask(self, quality_bitmask):
+
+    def _quality_mask(self, bitmask, mask):
         """Returns a boolean mask which flags all good-quality cadences.
 
         Parameters
         ----------
         quality_bitmask : int
             Bitmask. See ref. [1], table 2-3.
         """
-        return (self.hdu[1].data['SAP_QUALITY'] & quality_bitmask) == 0
+        bitmask_dict = {'default': KeplerQualityFlags.DEFAULT_BITMASK,
+                'conservative': KeplerQualityFlags.CONSERVATIVE_BITMASK,
+                'hard': KeplerQualityFlags.QUALITY_ZERO_BITMASK,
```

Comment:
If we are going to call this `hard`, then let's rename `KeplerQualityFlags.QUALITY_ZERO_BITMASK` to `KeplerQualityFlags.QUALITY_HARD_BITMASK`?

Reply:
I quite like ``KeplerQualityFlags.QUALITY_ZERO_BITMASK`` because it's easier to relate that with ``QUALITY==0``

Reply:
Ok!  I don't feel strongly about this.

Reply:
I'm happy taking input about whether these names should change? Currently we have 'default','hard','conservative','none'